test/unit/dataProvider/SyncDataProviderCollectionTest.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA ;
+ */
+
+namespace oat\taoSync\test\unit\dataProvider;
+
+use oat\generis\test\TestCase;
+use oat\taoSync\model\dataProvider\AbstractDataProvider;
+use oat\taoSync\model\dataProvider\SyncDataProviderCollection;
+use oat\taoSync\model\Exception\SyncDataProviderException;
+
+class SyncDataProviderCollectionTest extends TestCase
+{
+    public function testGetProvider()
+    {
+        $dataProvider = $this->createMock(AbstractDataProvider::class);
+        $dataProvider->method('getType')->willReturn('test');
+        $syncDataProviderCollection = new SyncDataProviderCollection(
+            [
+                SyncDataProviderCollection::OPTION_DATA_PROVIDERS => ['test' => $dataProvider]
+            ]
+        );
+        $syncDataProviderCollection->setServiceLocator($this->getServiceLocatorMock());
+
+        $this->assertEquals($dataProvider, $syncDataProviderCollection->getProvider('test'));
+    }
+
+    public function testGetInvalidProvider()
+    {
+        $syncDataProviderCollection = new SyncDataProviderCollection(
+            [
+                SyncDataProviderCollection::OPTION_DATA_PROVIDERS => [
+                    'dp' => $this->createMock(AbstractDataProvider::class)
+                ]
+            ]
+        );
+        $syncDataProviderCollection->setServiceLocator($this->getServiceLocatorMock());
+        $this->expectException(SyncDataProviderException::class);
+        $syncDataProviderCollection->getProvider('test');
+    }
+
+    public function testGetDataTest()
+    {
+        $dataProvider1 = $this->getMockBuilder(AbstractDataProvider::class)
```

Comment:
Can we extract test data initialization to dataProviders or helper methods where possible to make this test more readable?

Reply:
creating data provider mocks moved to separate method